src/configSchema.ts

```diff
@@ -464,18 +501,42 @@ export const VALIDATION_SCHEMA = z
 			});
 		}
 		if (
-			config.transmissionRpcUrl &&
+			(config.transmissionRpcUrl || config.rtorrentRpcUrl) &&
 			config.blockList.some((b) =>
 				b.startsWith(`${BlocklistType.CATEGORY}:`),
 			)
 		) {
 			logger.error(
-				`Transmission does not support ${BlocklistType.CATEGORY}: blocklisting, use ${BlocklistType.TAG}: instead for labels.`,
+				`Your client does not support ${BlocklistType.CATEGORY}: blocklisting, use ${BlocklistType.TAG}: instead for labels.`,
 			);
 			return false;
 		}
 		return true;
 	})
+	.refine(
+		(config) =>
+			!config.blockList.some(
+				(b) =>
+					b.startsWith(`${BlocklistType.CATEGORY}:`) ||
+					b.startsWith(`${BlocklistType.TAG}:`) ||
+					b.startsWith(`${BlocklistType.TRACKER}:`),
+			) ||
+			(config.torrentDir &&
+				(config.qbittorrentUrl ||
+					config.delugeRpcUrl ||
+					config.transmissionRpcUrl ||
+					config.rtorrentRpcUrl)),
+		ZodErrorMessages.blocklistNeedsClient,
+	)
+	.refine(
+		(config) =>
+			!config.blockList.some(
+				(b) =>
+					b.startsWith(`${BlocklistType.FOLDER}:`) ||
+					b.startsWith(`${BlocklistType.FOLDER_REGEX}:`),
+			) || config.dataDirs?.length,
+		ZodErrorMessages.blocklistNeedsDataDirs,
```

Comment:
can we add these as warning and promote to error in v7? right now they're not really harming anything and accidentally cross-seeding a couple extra things you didn't want is probably preferable to crash looping